lace/emulator/gp_emulator.py

```diff
@@ -147,17 +149,17 @@ def _buildTrainingSets(self,archive,paramList):
             print("Unknown emulator type, terminating")
             quit()
 
-        for aa in range(len(self.archive.data)):
+        for aa in range(len(getattr(self.archive, self.key_data))):
             for bb in range(len(paramList)):
-                params[aa][bb]=archive.data[aa][paramList[bb]] ## Populate parameter grid
+                params[aa][bb]=getattr(self.archive, self.key_data)[aa][paramList[bb]] ## Populate parameter grid
 
         return params,trainingPoints
 
 
     def _fit_p1d_in_archive(self,deg,kmax_Mpc):
         """For each entry in archive, fit polynomial to log(p1d)"""
         
-        for entry in self.archive.data:
+        for entry in getattr(self.archive, self.key_data):
```

Comment:
We have this structure in many places. Should we have a self.get_data() function instead?

Reply:
@lauracabayol - would this be a useful thing to do?